maker/src/connection.rs

```diff
@@ -39,13 +39,18 @@ use xtras::SendInterval;
 
 const TCP_TIMEOUT: Duration = Duration::from_secs(10);
 
-fn parse_vergen_version(vergen_string: &str) -> semver::Version {
-    let vergen = semver::Version::parse(vergen_string).expect("daemon version to parse");
+fn parse_vergen_version(vergen_string: &str) -> Result<semver::Version> {
+    let vergen = semver::Version::parse(vergen_string)
+        .context("Could not parse daemon_version: {vergen_string}")?;
```

Comment:
🔧 Ran into this myself recently, it's easy to miss:

```suggestion
        .with_context(|| format!("Could not parse daemon_version: {vergen_string}"))?;
```

Otherwise you will print the static string `"{vergen_string}"`.

Reply:
haha, good find 😅

Reply:
omg, that is indeed easy to miss!